pkg/client/clientcmd/validation.go

```diff
@@ -109,7 +109,8 @@ func validateClusterInfo(clusterName string, clusterInfo clientcmdapi.Cluster) [
 	if len(clusterInfo.Server) == 0 {
 		validationErrors = append(validationErrors, fmt.Errorf("no server found for %v", clusterName))
 	}
-	if len(clusterInfo.CertificateAuthority) != 0 {
+	// Only validate the CA file if it hasn't been overridden by CA data
```

Comment:
Is there a reason not to do it in the validateAuthInfo method of the validation.go file that is use to validate that a .kubeconfig is properly formed, right where we're commenting?